### PR TITLE
Adds VFS deleteFile method

### LINKED ISSUE #2799: missing `deleteFile` function in VirtualTypeScriptEnvironment
the VirtualTypeScriptEnvironment interface has two file-related functions `createFile` and `updateFile`, but seemingly no `deleteFile`.

https://github.com/microsoft/TypeScript-Website/blob/b89d197063b6849bbfc2edfb033f6082be950636/packages/typescript-vfs/src/index.ts#L22-L23

is this intended?
i've tried using the `VirtualTypeScriptEnvironment.sys.deleteFile` function, but it is undefined.  

my current workaround is to set the file contents to a single-space string (completely empty strings currently throw errors as reported here: #2713).  
it's not too big an issue for me right now since my virtual files are not directly visible and their names are auto-incremented, but i figured this would be worth posting here.

### PR DESCRIPTION
**Pull request by marklundin**
This PR adds a `deleteFile` function to `typescript/vfs` which works alongside the `createFile` and `updateFile` methods.

A test for is this has also been included.

Fixes #2799

### CODE REVIEW

**Comment by marklundin** [comment]
@microsoft-github-policy-service agree

**Comment by marklundin** [comment]
> @marklundin please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Review by kpal81xd — APPROVED** [review]

File: .changeset/tame-garlics-shop.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (jakebailey) [inline .changeset/tame-garlics-shop.md:1]: You shouldn't use an empty changeset here; this is a change that requires a package release with a bump.

**Review by jakebailey — CHANGES_REQUESTED** [review]
See previous comment; this isn't a dev only change and requires a proper versions changset and description.

File: .changeset/tame-garlics-shop.md (hunk 1 above)
Comment (marklundin) [reply .changeset/tame-garlics-shop.md:1]: Yep thanks @jakebailey. Have updated now

File: packages/typescript-vfs/src/index.ts [hunk 2]
```diff
@@ -99,6 +100,13 @@ export function createVirtualTypeScriptEnvironment(
 
       updateFile(newSourceFile)
     },
+    deleteFile(fileName) {
+      const sourceFile = languageService.getProgram()!.getSourceFile(fileName)
+      if (!sourceFile) {
+        throw new Error("Did not find a source file for " + fileName)
+      }
+      deleteFile(sourceFile)
```
Comment (jakebailey) [inline packages/typescript-vfs/src/index.ts]: Not sure either way, but does it make sense to throw on delete of a file that doesn't exist? Or should this just be a noop? The canonical implementation in TS itself catches errors, so never observes Node's throws for a nonexistent file.

Comment (marklundin) [reply packages/typescript-vfs/src/index.ts]: I don't have a strong opinion either way. For updateFile it make sense but a nooop here is probably acceptable

Comment (jakebailey) [reply packages/typescript-vfs/src/index.ts]: Yeah, I would probably make this operation not throw.

Comment (marklundin) [reply packages/typescript-vfs/src/index.ts]: Removed 👍

Comment (jakebailey) [reply packages/typescript-vfs/src/index.ts]: Thanks!

**Review by jakebailey — APPROVED** [review]

**Pull request merged**